tapcfg/config.go

```diff
@@ -257,12 +257,14 @@ type LndConfig struct {
 
 // UniverseConfig is the config that houses any Universe related config
 // values.
+//
+// nolint: lll
 type UniverseConfig struct {
 	SyncInterval time.Duration `long:"syncinterval" description:"Amount of time to wait between universe syncs"`
 
 	FederationServers []string `long:"federationserver" description:"The host:port of a Universe server peer with. These servers will be added as the default set of federation servers. Can be specified multiple times."`
 
-	PublicAccess bool `long:"public-access" description:"If true, and the Universe server is on a public interface, valid proof from remote parties will be accepted, and proofs will be queryable by remote parties. This applies to federation syncing as well as RPC insert and query."`
+	PublicAccess *string `long:"public-access" description:"The public access mode for the universe server. This can be unset, 'r', 'w', or 'rw'. If unset, public access is not enabled for the universe server. If 'r' is included, public access is allowed for read-only endpoints. If 'w' is included, public access is allowed for write endpoints."`
```

Comment:
Unfortunately using a pointer here doesn't have the desired effect of not requiring a value at all. If I just use the flag as it was set before (simulating an existing node) like `--universe.public-access`, I get the error message `expected argument for flag `--universe.public-access', but got option `--allow-public-uni-proof-courier'
failed to load config: expected argument for flag `--universe.public-access', but got option `--allow-public-uni-proof-courier'`

So if I explicitly wanted to pass an empty value, I'd need to set `--universe.public-access=""`, which will cause the argument not to be `nil` but an empty string, but then gives me the error `error creating server: unable to generate server config: unable to parse universe public access status: unknown universe public access status:`.

So I think we can just make this a string and just check for empty string. Too bad, would've been a nice way to remain backward compatible.

Reply:
Also, should we keep some of the explanation of what this flag actually influences from the original message?
Perhaps something along the lines of:
```go
"The public access mode for the universe server, controlling whether remote parties can read from and/or write to this universe server over RPC if exposed to a public network interface. This can be unset, 'r', 'w', or 'rw'. If unset, public access is not enabled for the universe server. If 'r' is included, public access is allowed for read-only endpoints. If 'w' is included, public access is allowed for write endpoints.
```